Documentation/Design/.NET Tool v2.md

```diff
@@ -0,0 +1,71 @@
+# Functional design .NET tool 2.0 `coverlet`
+
+Tracking issue https://github.com/tonerdo/coverlet/issues/683
+
+The idea is to support some new verbs:
+
+# coverlet run
+
+This will support current instrumentation workflow
+
+```
+coverlet run /path/to/test-assembly.dll --target "dotnet" --targetargs "test /path/to/test-project --no-build"
+```
+
+# coverlet mergereports
+
+Merge `json` default coverlet reports because we should otherwise implement merge for every supported format a huge effort for first version and also with the hope to integrate this feature directly inside collectors.
+
+We'll add an alias to `json` format called `coverlet` to distiguish from possible other future json format and also because this report make sense only inside specific test lifecycle.
+The content of this report is releated to internal structure and was not born to be "portable" or "stable" over time.
+
+```
+coverlet mergereports --coverletreport .\**\coverlet.json --targetdir .\Report\coverlet.xml --format cobertura
+```
+
+`--coverletreport`: will support glob pattern filtering  
+`--format`: specify a supported format  
+`--targetDir`: will create directory folder if not present. If specified a path that ends without a filename(no extension specified) we'll create a filename in that directory called `coverage.reportextension`, i.e.  
+
+`--format cobertura --targetDir .\Report` will create `.\Report\coverage.xml`  
+`--format cobertura --targetDir .\Report\MergedCoverage.xml` will create `.\Report\MergedCoverage.xml`  
+`--format cobertura --targetDir .\Report\MergedCoverage` will create `.\Report\MergedCoverage\coverage.xml`
+
+# coverlet validate 
+
+This command do validation on `json` coverlet format, because the structure is internal and we can support a lot of control without replicate alg for every supported format
+
+Validate could support in future different type of validations, for now we support threshold.
+
+```
+coverlet validate --type threshold --... --Fail or --Warn
```

Comment:
This need to specify which coverlet JSON files to process, which could be command line arguments in the same way as for mergereports. If multiple files are listed (or globbed) they should be merged before calculating the validation (but without the merge result being saved) to avoid having to first run a merge-to-coverlet followed by validate.

Reply:
yep you're right good catch...we could merge in memory and do the check